File: 1305.两棵二叉搜索树中的所有元素/1305.两棵二叉搜索树中的所有元素.py
'''
Descroption: LeetCode 1305. 两棵二叉搜索树中的所有元素
Author: EmoryHuang
Date: 2021-07-19 19:26:15
解题思路:
排序 + 遍历
'''


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
        ans = []

        def inorder(root: TreeNode):
            if not root:
                return
            inorder(root.left)
            ans.append(root.val)
            inorder(root.right)

        inorder(root1)
        inorder(root2)
        ans.sort()
        return ans

File: Solution/1305.两棵二叉搜索树中的所有元素/1305.两棵二叉搜索树中的所有元素.cpp
/*
 * @Descroption: LeetCode 1305. 两棵二叉搜索树中的所有元素
 * @Author: EmoryHuang
 * @Date: 2021-07-19 19:29:20
 * @解题思路:
 * 排序 + 遍历
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
   public:
    vector<int> ans;
    void inorder(TreeNode* root) {
        if (!root) return;
        inorder(root->left);
        ans.push_back(root->val);
        inorder(root->right);
    }
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        inorder(root1);
        inorder(root2);
        sort(ans.begin(), ans.end());
        return ans;
    }
};